### PR TITLE
Docs: structure + updates

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Documentation updates

### Details and comments

TODOs based on user feedback:
- [x] About Quantum Serverless
- [x] Installation
- [x] Preinstallation
  - [x] `pip install quantum-serverless` in your local environment
  - [x] If you want to deploy on your own infrastructure, link out to deployment info
- [x] Getting Started
  - [x] Running a program remotely. this only shows how to create program, save results, use jupyter widget to visualize job status, placeholder for credentials “token=…”, “host=…” for remote infrastructure, “warning” that everything in your working directory is shipped out remotely
  - [x] Running your program (reorder examples to the following:)
  - [x] Passing input arguments to your program
  - [x] Using python packages with programs
  - [x] Parallel workflows in programs 
  - [x] Retrieving results from old programs
- [x] Example programs
  - [x] VQE
    - [x] link to the source program file from the notebook
  - [x] QAOA
    - [x] link to the source program file from the notebook
    - [x] this is missing the final Sampler call to get the optimal solution. [See this example](https://qiskit.org/ecosystem/ibm-runtime/tutorials/qaoa_with_primitives.html#Solution-to-max-cut).
- [x] Migration guides
  - [x] Converting from Qiskit Runtime programs
- [x] Deployment

### CODE REVIEW

**Comment by IceKhan13** [comment]
Whaaaa?! 🤔 

```
Error response from daemon: Get "https://registry-1.docker.io/v2/": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

**Comment by psschwei** [comment]
that happens very infrequently (usually a network flake or flicker)... I just reran the test and I'd assume this time should be fine (:crossed_fingers: )

**Comment by psschwei** [comment]
hmm... Just had it fail pulling r4 locally, but then succeeded the next time...

```
$ docker pull bitnami/keycloak:20.0.5-debian-11-r4
20.0.5-debian-11-r4: Pulling from bitnami/keycloak
Get "https://registry-1.docker.io/v2/bitnami/keycloak/manifests/sha256:3cd9558d458d583e0c3b22918f5b194d75d2d89c2091d68b06e6d74ec4a5bda9": EOF

$ docker pull bitnami/keycloak:20.0.5-debian-11-r4
20.0.5-debian-11-r4: Pulling from bitnami/keycloak
Digest: sha256:f0b821c44296b2461331cce1b0abbbff9c284bb6a037b8ccde8bbea4fe23ec3f
Status: Downloaded newer image for bitnami/keycloak:20.0.5-debian-11-r4
docker.io/bitnami/keycloak:20.0.5-debian-11-r4
```

**Comment by psschwei** [comment]
btw, this images is ooooooold (`20.0.5-debian-11-r4` was last pushed 7 months ago...)

**Comment by caleb-johnson** [comment]
I'm looking through the built docs as I'm reviewing, and I think I feel strongly that the `Running` section should be merged into `Getting Started`. `Getting Started` can just be the collection of these simple notebooks that feature some basic, fundamental feature of Serverless.

Still going through everything, but thought this deserved its own note.

**Comment by IceKhan13** [comment]
hm , looks like docker registry is not feeling well today 😷 

Ok, I will let it sleep tonight and check tomorrow morning :)

**Comment by IceKhan13** [comment]
> I'm looking through the built docs as I'm reviewing, and I think I feel strongly that the Running section should be merged into Getting Started. Getting Started can just be the collection of these simple notebooks that feature some basic, fundamental feature of Serverless.

@caleb-johnson let me know if I need to merge those sections. I have no strong feeling about it. It was just in a feedback doc

**Comment by psschwei** [comment]
seems dockerhub is back to playing nice again :smiley:

**Comment by caleb-johnson** [comment]
> I'm looking through the built docs as I'm reviewing, and I think I feel strongly that the `Running` section should be merged into `Getting Started`. `Getting Started` can just be the collection of these simple notebooks that feature some basic, fundamental feature of Serverless.
> 
> I find the term `Running` to be vague, and I think merging this leans up things for the better.
> 
> Still going through everything, but thought this deserved its own note.

@jenglick , any thoughts on merging these 2 sections?

**Comment by caleb-johnson** [comment]
@IceKhan13 , do we have a Quantum Serverless logo? It'd be nice to have *something* for the thumbnails in our docs. The "NO THUMBNAIL" is just too ugly :P

**Comment by caleb-johnson** [comment]
```
provider = Provider(
    username=os.environ.get("GATEWAY_USER", "user"),
    password=os.environ.get("GATEWAY_PASSWORD", "password123"),
    # token=os.environ.get("GATEWAY_TOKEN", "<TOKEN>"), # token can be used instead of user/password combination
    host=os.environ.get("GATEWAY_HOST", "http://localhost:8000/"),
)

----------------

| Message: Http bad request.
| Code: 400
```
I'm running `jupyter` docker profile and run running/notebooks/01_...  and getting an HTTP request?

@IceKhan13 , should this notebook run under the `jupyter` profile available?

**Comment by jenglick** [comment]
> > I'm looking through the built docs as I'm reviewing, and I think I feel strongly that the `Running` section should be merged into `Getting Started`. `Getting Started` can just be the collection of these simple notebooks that feature some basic, fundamental feature of Serverless.
> > I find the term `Running` to be vague, and I think merging this leans up things for the better.
> > Still going through everything, but thought this deserved its own note.
> 
> @jenglick , any thoughts on merging these 2 sections?

agreed!

**Comment by IceKhan13** [comment]
@caleb-johnson your PR merged. Let me know if we can merge this?

**Comment by caleb-johnson** [comment]
> @caleb-johnson your PR #986 merged. Let me know if we can merge this?

I'm going to submit 1-2 more PRs to address @jenglick feedback, and then we can merge.

**Comment by caleb-johnson** [comment]
@IceKhan13 , any idea why `serverless/getting_started/experimental/experimental_running_programs_using_decorators.ipynb` cannot find the module `serverless/getting_started/experimental/source_files/circuit_utils.py` in the CI? 

This works for me locally. The only thing I can think is that maybe it is looking in some other `source_files` directory?

File: docs/examples/examples/source_files/vqe/vqe.py [hunk 1]
```diff
@@ -145,10 +143,27 @@ def run_vqe(
...
+        sampler = QiskitSampler()
+    samp_dist = sampler.run(qc, shots=int(1e4)).result().quasi_dists[0]
+
     save_result({
+        "result": samp_dist,
         "optimal_point": vqe_result.x.tolist(),
         "optimal_value": vqe_result.fun,
-        "optimizer_evals": vqe_result.nfev,
-        "optimizer_history": callback_dict.get("cost_history", []),
+        # "optimizer_evals": vqe_result.nfev,
```
[the remark this answers is not in the document]
Comment (IceKhan13) [reply docs/examples/examples/source_files/vqe/vqe.py]: results are too bulky with those values + we do not need them

**Pull request merged**